libraries/AP_HAL/SPIDevice.h

```diff
@@ -66,6 +66,12 @@ class SPIDeviceManager {
     {
         return nullptr;
     }
+
+    /* Return spi count number table. */
+    virtual uint8_t get_count() { return 0; }
```

Comment:
SPIDeviceManager is more than just a table with **possible** devices. It actually manages the detected devices.  This name is confusing because it doesn't specify if it's returning "active/detected/probed" devices or just the list of all possible devices for this board. I know it's returning the latter... but I think that should be internal to the spi device implementation and not exposed.